gensim/test/test_ldamodel.py

```diff
@@ -508,6 +508,37 @@ def testAlphaAuto(self):
 # endclass TestLdaMulticore
 
 
+class TestLdaIterableTraining(unittest.TestCase, basetmtests.TestBaseTopicModel):
+    """Class for testing infinite generators,
+    for case when size of processed documents
+    goes beyond corpus length."""
+
+    class ProblematicIterable:
```

Comment:
There's no benefit to nesting classes here. Please move this new class to module scope.